### PR TITLE
chore(repo): clean up repo and convert workflow

### PR DESCRIPTION
**Pull request by rxmarbles**
This PR accomplishes a few things

1. Converts workflows to 1 workflow but handles all usecases (this also fixes the release action failing as it was trying to publish on each job
2. Switches from `master` to `main` to be more progressive and aligned w/ our other repositories
3. Updates README w/ a new badge for GHA and a bit of Markdown lint cleanup
4. Fixes moderate security issue w/ `semantic-release`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   "scripts": {
     "test-typings": "tsc --lib ES2015 --noEmit typings/*.ts",
     "test": "mocha index.spec.js lib/**/*.spec.js && npm run test-typings",
+    "posttest": "npm run lint",
```
Comment (gergelyke-dd) [inline package.json:10]: any reason for not running it as part of the test script above?

**Review by gergelyke-dd — APPROVED** [review]
lgtm

File: package.json (hunk 1 above)
Comment (rxmarbles) [reply package.json:10]: I don't have any strong opinions not to. Which ever you prefer honestly 😄. Most patterns I've seen is handling it in the `posttest` cycle vs. in the main `test` cycle` We could do this `pretest` as well to fail early if you choose

**Review by gergelyke — APPROVED** [review]
please move linting under the test command, looks good otherwise

**Pull request merged**